dallinger/experiment.py

```diff
@@ -523,6 +521,16 @@ def make_uuid(cls):
         """Generate a new uuid."""
         return str(uuid.UUID(int=random.getrandbits(128)))
 
+    def _finish_experiment(self):
+        # Debug runs synchronously
+        if self.exp_config.get('mode') != 'debug':
+            self.log("Waiting for experiment to complete.", "")
+            while self.experiment_completed() is False:
+                time.sleep(30)
+            data = self.retrieve_data()
```

Comment:
This needs to be un-indented so that data is retrieved whether or not in debug mode, and the `return data` doesn't result in a `NameError`.